fu/ALUgen_Ron_2nd/ALUgenMAC2nd.py

```diff
@@ -0,0 +1,133 @@
+"""
+==========================================================================
+ALUgenMAC2nd.py
+==========================================================================
+Floating point add unit.
+
+Author : Yanghui Ou & Ron Jokai
+  Date : Jan 6, 2024
+"""
+
+from pymtl3                                import *
+from ...lib.ifcs                           import SendIfcRTL, RecvIfcRTL
+from ...lib.opt_type                       import *
+from ..basic.Fu                            import Fu
+from ..pymtl3_Ron_fixed.ALUgenMACRTL       import ALUgenMAC
+
+class ALUgenMAC2nd( Fu ):
+  def construct( s, DataType, PredicateType, CtrlType,
+                 num_inports, num_outports, data_mem_size):
+    super( ALUgenMAC2nd, s ).construct( DataType, PredicateType, CtrlType,
+                                   num_inports, num_outports,
+                                   data_mem_size )
+
+    # Local parameters
+    assert DataType.get_field_type( 'payload' ).nbits == 16
+
+    num_entries = 3
+    FuInType    = mk_bits( clog2( num_inports + 1 ) )
+    CountType   = mk_bits( clog2( num_entries + 1 ) )
+
+    # Components
+    s.fALU = ALUgenMAC()
+    s.fALU.rhs_2 //= lambda:   0 if s.recv_opt.msg.ctrl == OPT_ADD     else \
+                             ( 1 if s.recv_opt.msg.ctrl == OPT_SUB     else \
+                             ( 3 if s.recv_opt.msg.ctrl == OPT_LT      else \
+                             ( 7 if s.recv_opt.msg.ctrl == OPT_GTE     else \
+                             (10 if s.recv_opt.msg.ctrl == OPT_GT      else \
+                             (14 if s.recv_opt.msg.ctrl == OPT_LTE     else \
+                             (16 if s.recv_opt.msg.ctrl == OPT_MUL     else \
+                             (48 if s.recv_opt.msg.ctrl == OPT_MUL_ADD else 0)))))))
```

Comment:
Would the generated logic cause bad timing/long critical path? Do you think use if/else inside the update block would help? Or a smarter mux should be modeled and used?

Reply:
As you suggested, we can optimize this later. I would use an and/or MUX here since the select line is one-hot. I will need to ramp up on how basic RTL constructs are implemented in PyMTL.

Reply:
assign s.fALU.rhs_2 = (6'd0 & {6{s.recv_opt.msg.ctrl == OPT_ADD}} |
                                      (6'd1 & {6{s.recv_opt.msg.ctrl == OPT_SUB}} | ...

Reply:
nit: i suggest we don't use hardcoded `6` in the RTL module, as it would make it difficult to parameterize or later if we want to increase the number of opcodes.